cli/src/commands/branch.rs

```diff
@@ -301,6 +319,42 @@ fn cmd_branch_create(
     Ok(())
 }
 
+fn cmd_branch_rename(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchRenameArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let repo = workspace_command.repo().as_ref();
+    let old_branch = &args.old;
+    // call clone because .set_local_branch_target takes ownership
+    // of the target_commit, so we need to create a copy that is no
+    // longer owned by the workspace.
+    let target_commit = repo.view().get_local_branch(old_branch).clone();
+    if target_commit.is_absent() {
+        return Err(user_error_with_hint(
+            format!("Branch not found: {old_branch}"),
+            "Target branch name does not exist.",
+        ));
+    }
+
+    let new_branch = &args.new;
+    let mut tx = workspace_command.start_transaction();
+    tx.mut_repo()
+        .set_local_branch_target(new_branch, target_commit);
+    tx.mut_repo()
+        .set_local_branch_target(old_branch, RefTarget::absent());
+    tx.finish(
+        ui,
+        format!(
+            "renamed {} to {}",
```

Comment:
nit: `rename` (instead of `renamed`)

Reply:
Done.

For my own education, my thinking was that this was shown _after_ the rename was complete, hence the 'renamed'. Is that correct?

Reply:
It's the message displayed in `jj op log`, and we appear to use present tense.

Reply:
Ahh... This makes sense. Thanks.